modules/charon-core/src/main/java/org/wso2/charon3/core/extensions/RoleManager.java

```diff
@@ -0,0 +1,116 @@
+     * @throws CharonException         CharonException.
+     * @throws ConflictException       ConflictException.
+     * @throws NotImplementedException NotImplementedException.
+     * @throws BadRequestException     BadRequestException.
+     */
+    Role createRole(Role role) throws CharonException, ConflictException, NotImplementedException, BadRequestException;
+
+    /**
+     * Get the role for the given ID.
+     *
+     * @param id                 Role ID.
+     * @param requiredAttributes Required Attributes.
+     * @return Role.
+     * @throws NotImplementedException NotImplementedException.
+     * @throws BadRequestException     BadRequestException.
+     * @throws CharonException         CharonException.
+     * @throws NotFoundException       NotFoundException.
+     */
+    Role getRole(String id, Map<String, Boolean> requiredAttributes)
+            throws NotImplementedException, BadRequestException, CharonException, NotFoundException;
+
+    /**
+     * Delete the given role.
+     *
+     * @param id Role ID.
+     * @throws NotFoundException       NotFoundException.
+     * @throws CharonException         CharonException.
+     * @throws NotImplementedException NotImplementedException.
+     * @throws BadRequestException     BadRequestException.
+     */
+    void deleteRole(String id) throws NotFoundException, CharonException, NotImplementedException, BadRequestException;
+
+    /**
+     * List roles with Get.
+     *
+     * @param node       Node
+     * @param startIndex Start Index
+     * @param count      Count
+     * @param sortBy     Sort by
+     * @param sortOrder  Sort order
+     * @return List of roles.
+     * @throws CharonException         CharonException.
+     * @throws NotImplementedException NotImplementedException.
+     * @throws BadRequestException     BadRequestException.
+     */
+    List<Object> listRolesWithGET(Node node, Integer startIndex, Integer count, String sortBy, String sortOrder)
+            throws CharonException, NotImplementedException, BadRequestException;
+
+    /**
+     * Update the role.
+     *
+     * @param oldRole Old role.
+     * @param newRole new role.
+     * @return Updated role.
+     * @throws NotImplementedException NotImplementedException.
+     * @throws BadRequestException     BadRequestException.
+     * @throws CharonException         CharonException.
+     * @throws NotFoundException       NotFoundException.
+     */
+    Role updateRole(Role oldRole, Role newRole)
+            throws NotImplementedException, BadRequestException, CharonException, NotFoundException;
+
+    /**
+     * List roles with Post.
+     *
+     * @param searchRequest Search request.
+     * @return List of roles.
+     * @throws NotImplementedException NotImplementedException.
+     * @throws BadRequestException     BadRequestException.
+     * @throws CharonException         CharonException.
+     */
+    List<Object> listRolesWithPost(SearchRequest searchRequest)
```

Comment:
shouldn't we support pagination here?

Reply:
Pagination parameters are included inside the `searchRequest`.